htlcswitch/decayedlog_test.go

```diff
@@ -82,24 +71,19 @@ func startup(dbPath string, notifier bool) (sphinx.ReplayLog,
 	return log, chainNotifier, &hashedSecret, stop, nil
 }
 
-// shutdown deletes the temporary directory that the test database uses
-// and handles closing the database.
-func shutdown(dir string, d sphinx.ReplayLog) {
-	d.Stop()
-	os.RemoveAll(dir)
-}
-
 // TestDecayedLogGarbageCollector tests the ability of the garbage collector
 // to delete expired cltv values every time a block is received. Expired cltv
 // values are cltv values that are < current block height.
 func TestDecayedLogGarbageCollector(t *testing.T) {
 	t.Parallel()
 
-	dbPath := tempDecayedLogPath(t)
+	dbPath := t.TempDir()
 
 	d, notifier, hashedSecret, _, err := startup(dbPath, true)
 	require.NoError(t, err, "Unable to start up DecayedLog")
-	defer shutdown(dbPath, d)
+	t.Cleanup(func() {
+		_ = d.Stop()
```

Comment:
nit: check the error here and below too(`require.NoError`)?

Reply:
Hi @yyforyongyu, thanks for reviewing this PR. I've rebased and added the error checks. Please take a look again. Thanks